webpack/assets/javascripts/react_app/components/AutoComplete/AutoCompleteSelectors.js

```diff
@@ -0,0 +1,17 @@
+export const selectAutocomplete = state => state.autocomplete;
+
+export const selectAutocompleteError = state => selectAutocomplete(state).error;
+
+export const selectAutocompleteResults = state => selectAutocomplete(state).results.asMutable();
```

Comment:
Where do we actually break the `immutability`?
Can we change the consumer of the `results` array so it won't break the `immutability`?
If a 3rd party component breaks the `immutability`? If so can we pass the `results.asMutable()` only to the problematic component?